aspnetcore/fundamentals/logging/loggermessage.md

```diff
@@ -0,0 +1,173 @@
+      Quote deleted (Quote = 'You can avoid reality, but you cannot avoid the consequences of avoiding reality. - Ayn Rand' Id = 1)
+```
+
+You can inspect the message in the app's console output when the exception is logged:
+
+```console
+fail: LoggerMessageSample.Pages.IndexModel[5]
+      => RequestId:0HL90M6E7PHK5:00000010 RequestPath:/ => /Index
+      Quote delete failed (Id = 999)
+System.ArgumentNullException: Value cannot be null.
+Parameter name: entity
+   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
+   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
+   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Remove(TEntity entity)
+   at LoggerMessageSample.Pages.IndexModel.<OnPostDeleteQuoteAsync>d__14.MoveNext() in 
+      <PATH>\sample\Pages\Index.cshtml.cs:line 87
+```
+
+## Implementing LoggerMessage.DefineScope
+
+Define a [log scope](xref:fundamentals/logging/index#log-scopes) to apply to a series of log messages using the [DefineScope(String)](/dotnet/api/microsoft.extensions.logging.loggermessage.definescope) method.
+
+The sample app has a **Clear All** button for deleting all of the quotes in the database. The quotes are deleted by removing them one at a time. Each time a quote is deleted, the `QuoteDeleted` method is called on the logger. A log scope is added to these log messages.
+
+Enable `IncludeScopes` in the console logger options:
+
+```csharp
+logging.AddConsole(options => options.IncludeScopes = true);
+```
+
+Setting `IncludeScopes` is required in ASP.NET Core 2.0 apps to enable log scopes. Setting `IncludeScopes` via *appsettings* configuration files is a feature that will be available with the ASP.NET Core 2.1 release.
+
+The sample app also clears other providers and adds filters to reduce the logging output. This makes it easier to see the sample's log messages that demonstrate `LoggerMessage` features:
+
+[!code-csharp[Main](loggermessage/sample/Program.cs?name=snippet1&highlight=22)]
+
+Create a field to hold a `Func` delegate for the log scope. The sample app creates a field called `_allQuotesDeletedScope` (*Internal/LoggerExtensions.cs*):
+
+[!code-csharp[Main](loggermessage/sample/Internal/LoggerExtensions.cs?name=snippet4)]
+
+Use `DefineScope` to create the delegate. Up to three types can be specified for use as template arguments when the delegate is invoked. The sample app uses a message template that includes the number of quotes deleted (an `int` type):
+
+[!code-csharp[Main](loggermessage/sample/Internal/LoggerExtensions.cs?name=snippet8)]
+
+Provide a static extension method with parameters for the types you intend to provide in the log message and returns the `Func` you created. The sample app takes in a `count` of quotes to delete and returns `_allQuotesDeletedScope`:
+
+[!code-csharp[Main](loggermessage/sample/Internal/LoggerExtensions.cs?name=snippet12)]
```

Comment:
Looks like you don't actually show how to use the scope? Also the messages below don't seem to include the custom scope.